File: Pelikula.API/Model/Projekcija/ProjekcijaTerminResponse.cs
using System;

namespace Pelikula.API.Model.Projekcija
{
    public partial class ProjekcijaTerminResponse
    {
        public int Id { get; set; }
        public DateTime? Termin { get; set; }

        public ProjekcijaResponse Projekcija { get; set; }

        public override string ToString() {
            return $"{Projekcija} - {Termin.GetValueOrDefault(): dd/MM/yyyy, HH:mm}";
        }
    }
}

File: Pelikula.MOBILE/lib/model/projekcija/projekcija_termin_response.dart
import 'package:pelikula_mobile/model/projekcija/projekcija_response.dart';

class ProjekcijaTerminResponse {
  int? id;
  DateTime? termin;

  ProjekcijaResponse? projekcija;

  ProjekcijaTerminResponse({
    this.id,
    this.termin,
    this.projekcija,
  });

  factory ProjekcijaTerminResponse.fromJson(Map<String, dynamic> jsonObj) {
    return ProjekcijaTerminResponse(
      id: jsonObj['id'] as int,
      termin: DateTime.tryParse(jsonObj['termin']),
      projekcija: ProjekcijaResponse.fromJson(jsonObj['projekcija']),
    );
  }

  Map<String, dynamic> tojsonObj() => {
        "id": id,
        "termin": termin,
        "projekcija": projekcija,
      };
}
